DietPi-Software | Rclone

goldfix opened the pull request:
Added script to install (and remove) Rclone tool.

goldfix committed:
added scripts to add and remove Rclone

goldfix committed:
Merge branch 'dev' into rclone_v1

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -12138,6 +12144,15 @@ _EOF_
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Installing
+			G_EXEC curl -fsSL https://rclone.org/install.sh -o /tmp/rclone_install.sh
```
You could use our `Download_Install` function and download the zip archive directly from "-`https://downloads.rclone.org/rclone-current-linux-${OS_type}.zip` directly. Have a look to the Wordpress install (software_id=55) . An example to detect OS typ/architecture could be our Emby installation (software_id=41)

goldfix replied:
it's ok. It was bit more complicated because the Rclone's unzipped folder contains the "version".

Joulinar replied:
what do you think on following. It might simplify the way to detect the version

```
		software_id=202 # Rclone
		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then

			Banner_Installing

			# ARMv6/7
			local arch='arm'

			# ARMv8
			if (( $G_HW_ARCH == 3 )); then
				arch='arm64'

			# x86_64
			elif (( $G_HW_ARCH == 10 )); then
				arch='amd64'

			fi

			local version=$(curl -sSfL 'https://downloads.rclone.org/version.txt' | mawk '{print $2}')
			Download_Install "https://downloads.rclone.org/rclone-current-linux-$arch.zip"
			# - Reinstall: remove old executable 
			[[ -f '/usr/bin/rclone' ]] && G_EXEC rm /usr/bin/rclone
			# - Move new executable in place
			G_EXEC mv "rclone-$version-linux-$arch/rclone" /usr/bin/rclone
			# - Cleanup temp file system
			G_EXEC rm -Rf "rclone-$version-linux-$arch/rclone"

		fi
```

As well added some error handling `G_EXEC` and some comments 😃

Joulinar replied:
ah btw, do you know if Rclone is supporting ARMv6? eg RPi1? If not we might need to disable.

Joulinar replied:
seems to be working on ARMv6 as well 

```
root@DietPi1:~# rclone version
rclone v1.58.1
- os/version: debian 11.3
- os/kernel: 5.15.32+ (armv6l)
- os/type: linux
- os/arch: arm
- go/version: go1.17.9
- go/linking: static
- go/tags: none
root@DietPi1:~#
```

goldfix replied:
Hi @Joulinar , 
yes the version "arm" is compatible with ARM6 and ARM7 (tested on Raspberry Pi Zero 1 and 2).

Joulinar commented on dietpi/dietpi-software:
```diff
@@ -825,7 +825,13 @@ Available commands:
+		aSOFTWARE_NAME[$software_id]='Rclone'
+		aSOFTWARE_DESC[$software_id]='Rclone syncs your files to cloud storage'
+		aSOFTWARE_CATX[$software_id]=4
+		aSOFTWARE_DOCS[$software_id]='https://rclone.org/'
```
Docs should point to our own online Docs

goldfix committed:
fix link to docs

goldfix committed:
substituted rclone's install script with internal Download_Install function

goldfix committed:
refactoring install procedure

Joulinar committed:
v8.6

DietPi-Software | Rclone - simplify install and removal as we use *.deb package available on GitHub

Joulinar committed:
v8.6

DietPi-Software | Rclone - remove configuration files from user home directory during uninstall

Joulinar committed:
Merge branch 'dev' into rclone_v1

MichaIng committed:
v8.6

- DietPi-Software | Rclone: Add meta info

MichaIng committed:
v8.6

- DietPi-Software | Rclone: Use arm-v7 binary on ARMv7
